modules/git/repo_branch_nogogit.go

```diff
@@ -107,9 +107,9 @@ func walkShowRef(ctx context.Context, repoPath, arg string, skip, limit int, wal
 
 	go func() {
 		stderrBuilder := &strings.Builder{}
-		args := []string{"show-ref"}
+		args := []string{"for-each-ref", "--format=%(objectname) %(refname)"}
 		if arg != "" {
-			args = append(args, arg)
+			args = append(args, strings.Fields(arg)...)
```

Comment:
Splitting arguments by space is dangerous.
* https://github.com/go-gitea/gitea/pull/21168